pkg/cache/memory/repository.go

```diff
@@ -266,6 +267,54 @@ func (r *cachedRepository) update(ctx context.Context, updated repository.Packag
 	// TODO: Just updated package?
 	identifyLatestRevisions(r.cachedPackageRevisions)
 
+	if updated.Lifecycle() == (v1alpha1.PackageRevisionLifecyclePublished) {
+		updatedMain := updated.ToMainPackageRevision()
+		//Search and delete any old main pkgRev in the cache
+		for k := range r.cachedPackageRevisions {
+			if (k.Repository == updatedMain.Key().Repository) && (k.Package == updatedMain.Key().Package) && (k.Revision == updatedMain.Key().Revision) {
+				oldMainKey := repository.PackageRevisionKey{
+					Repository:    updatedMain.Key().Repository,
+					Package:       updatedMain.Key().Package,
+					Revision:      updatedMain.Key().Revision,
+					WorkspaceName: v1alpha1.WorkspaceName(string(k.WorkspaceName)),
+				}
+				delete(r.cachedPackageRevisions, oldMainKey)
+			}
+		}
+		cachedMain := &cachedPackageRevision{PackageRevision: updatedMain}
+		r.cachedPackageRevisions[updatedMain.Key()] = cachedMain
+
+		pkgRevMetaNN := types.NamespacedName{
+			Name:      updatedMain.KubeObjectName(),
+			Namespace: updatedMain.KubeObjectNamespace(),
+		}
+		// TODO: error handling for metadataStore.Get
+		_, err := r.metadataStore.Get(ctx, pkgRevMetaNN)
+		// Error can be different
```

Comment:
Maybe write a more meaningful comment?

Reply:
Sorry, my bad, this was just for me, I'll remove the comment.